src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpCertificateHelper.cs

```diff
@@ -74,5 +75,50 @@ internal static class CertificateHelper
                 }
             }
         }
+
+        public static X509Certificate2 GetEligibleClientCertificate()
+        {
+            // Get initial list of client certificates from the MY store.
+            X509Certificate2Collection candidateCerts;
+            using (var myStore = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            {
+                myStore.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
+                candidateCerts = myStore.Certificates;
+            }
+            
+            return GetEligibleClientCertificate(candidateCerts);
+        }
+        
+        public static X509Certificate2 GetEligibleClientCertificate(X509Certificate2Collection candidateCerts)
+        {
+            if (candidateCerts.Count == 0)
+            {
+                return null;
+            }
+
+            // Reduce the set of certificates to match the proper 'Client Authentication' criteria.
+            candidateCerts = candidateCerts.Find(X509FindType.FindByKeyUsage, X509KeyUsageFlags.DigitalSignature, false);
+            candidateCerts = candidateCerts.Find(X509FindType.FindByApplicationPolicy, ClientAuthenticationOID, false);
+
+            // Build a new collection with certs that have a private key. Need to do this
+            // manually because there is no X509FindType to match this criteria.
+            var eligibleCerts = new X509Certificate2Collection();
+            foreach (var cert in candidateCerts)
+            {
+                if (cert.HasPrivateKey)
+                {
+                    eligibleCerts.Add(cert);
```

Comment:
If below question's answer is that returning the first one is correct, then we should stop after we've found one and return immediately instead of continuing the search.